chips/imxrt10xx/src/ccm_analog.rs

```diff
@@ -0,0 +1,625 @@
+//! CCM Analog peripheral
+
+// Generated following
+//
+// 1. Use svd2regs.py to import the memory layout and all fields
+// 2. For each reg, set, clear, toggle, grouping, replace it with
+//    a Groups struct.
+// 3. Remove unused
+
+use kernel::common::registers::{
+    register_bitfields, register_structs, ReadWrite, RegisterLongName, WriteOnly,
+};
+use kernel::common::StaticRef;
+
+/// Many CCM_ANALOG registers are laid out with additional set, clear, and toggle
+/// registers. This groups them together into an extended register.
+///
+/// Selecting suffix 'SCT' given the mention in the reference manual:
+///
+/// > A set of SCT registeres is offered for registers in many modules [...]
+///
+/// Although the CCM_ANALOG memory map indicates the SCT registers are R/W,
+/// the reference manual front-matter (section 1.5.2) indicates that they should
+/// be treated as as write only:
+///
+/// > The SCT registers always read back 0, and should be considered write-only.
+#[repr(C)]
+struct RegisterSCT<R: RegisterLongName = ()> {
+    /// The normal register
+    reg: ReadWrite<u32, R>,
+    /// Write 1 sets bits in reg
+    set: WriteOnly<u32, R>,
+    /// Write 1 clears bits in reg
+    clear: WriteOnly<u32, R>,
+    /// Write 1 toggles bits in reg
+    toggle: WriteOnly<u32, R>,
+}
+
+register_structs! {
+    /// CCM_ANALOG
+    CcmAnalogRegisters {
+        /// Analog ARM PLL control Register
+        (0x000 => pll_arm: RegisterSCT<PLL_ARM::Register>),
+        /// Analog USB1 480MHz PLL Control Register
+        (0x010 => pll_usb1: RegisterSCT<PLL_USB1::Register>),
+        /// Analog USB2 480MHz PLL Control Register
+        (0x020 => pll_usb2: RegisterSCT<PLL_USB2::Register>),
+        /// Analog System PLL Control Register
+        (0x030 => pll_sys: RegisterSCT<PLL_SYS::Register>),
+        /// 528MHz System PLL Spread Spectrum Register
+        (0x040 => pll_sys_ss: ReadWrite<u32, PLL_SYS_SS::Register>),
+        (0x044 => _reserved0),
```

Comment:
Same here.

Reply:
@lschuermann are you sure about this? I don't think the macro will output the reserved space without this.

```
(0x048 => clk_peri_ctrl: ReadWrite<u32, CLK_PERI_CTRL::Register>),
// (0x04C => _reserved0),
/// Indicates which src is currently selected (one-hot)
(0x050 => clk_peri_selected: ReadOnly<u32, CLK_PERI_SELECTED::Register>),
```
produces
```
#[doc =
    r" Clock control, can be changed on-the-fly (except for auxsrc)"]
clk_peri_ctrl: ReadWrite<u32, CLK_PERI_CTRL::Register>,
#[doc = r" Indicates which src is currently selected (one-hot)"]
clk_peri_selected: ReadOnly<u32, CLK_PERI_SELECTED::Register>,
```

Reply:
I am not actually, I've never used the syntax myself! I actually didn't see the point in the syntax if it _wouldn't_ work, but given the macro output you're obviously right. Thanks for the hint!

Reply:
Yeah, sorry, I've just now realized my mistake: the syntax acutally does make sense, because it automatically sizes the reserved fields correctly, though doesn't get rid of them entirely. `proc_macros` probably aren't smart enough to autogenerate the reserved field names (except maybe with some really clever ident concat hacks which we probably don't really want to use anyways) :smile:.